### PR TITLE
Fixes #211 Handle non-existent repositories in WebUI

### LINKED ISSUE #211: Handle non-existent repositories is yaydoc WebUI
### This is a
- [x] Bug report
- [ ] Feature request  

### Description

There are some unexpected behaviours when a repository with the inputted URL does not exist. Handle this by failing fast

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Using `git ls-remote ${GITURL}` with Github repositories require us to input username and password. To avoid that, Github API is used
## Related Issue

issue: #211 

## Screenshot
image

## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by pri22296** [comment]
@imujjwal96 Are you sure git ls-remote requires username and password? I just ran it it didn't prompt for one!

**Comment by imujjwal96** [comment]
image

Cloning a non-existent repository
image

**Comment by pri22296** [comment]
For `git clone` instead of `>/dev/null 2>&1`, could you try the -q switch?

**Comment by imujjwal96** [comment]
image

**Comment by pri22296** [comment]
@imujjwal96 It seems that when `git clone` and `git ls-remote` cannot find a repo, It assumes it's a private one and asks for a password. we could use the `username:password@` syntax and it wouldn't prompt!

**Comment by imujjwal96** [comment]
Why should we do this?

**Comment by pri22296** [comment]
I guess there's no real advantage of using one approach over the other.

**Comment by imujjwal96** [comment]
No, i mean for this we'll have to ask the user's password, right?

**Comment by pri22296** [comment]
Oh, well since that username and password would only be needed when the repo doesn't even exists, does it even matter what username or password we provide. for all i know, we could `git clone -q https://:@github.com/pri22296/nonrepo.git` and it would no longer prompt for a password.

**Comment by imujjwal96** [comment]
Do you want me to change the value of GITURL then?

**Comment by pri22296** [comment]
I don't really have a preference here. what do you think which way would be better? The only merit I can think of with changing the giturl that it would certainly lead to less code as compared to the current approach.

**Comment by imujjwal96** [comment]
@pri22296  I've updated it. Thanks for your valuable suggestions :)

File: generate.sh [hunk 1]
```diff
@@ -27,20 +34,19 @@ if [ "${WEBUI:-false}" == "true" ]; then
   else
     mkdir -p temp/${EMAIL} && cd $_
     echo -e "Cloning Repository...\n"
-    git clone -q ${GITURL} "$UNIQUEID" && cd $_
+    git clone -q https://:@${PLATFORM}/${USERNAME}/${REPONAME} "$UNIQUEID" >/dev/null 2>&1
+    if [ $? -ne 0 ]; then
+      >&2 echo "Failed to Clone. Repository does not exist."
```
Comment (pri22296) [inline generate.sh]: add a `\n` at the end of the message? would be consistent with other error messages.

Comment (imujjwal96) [inline generate.sh]: @pri22296  updated

**Comment by imujjwal96** [comment]
@sch00lb0y  Please review,

**Pull request merged**